### ISSUE TITLE
JavaNetClientBuilder fails with NoContent response

### ISSUE THREAD

**Issue by steve-e** [http4s/http4s#2114]
Calling delete on a service that returns NoContent results in an exception when using JavaNetClientBuilder.
The call is 
```
httpClient.status(Request[IO](Method.DELETE, uri))
```
The exception I get is 
```
Caused by: java.io.FileNotFoundException: http://localhost:8080/history/98613681-91f2-4f42-ac23-655de09178b3
        ...
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1490)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
	at org.http4s.client.JavaNetClientBuilder.$anonfun$open$21(JavaNetClientBuilder.scala:120)
       ...
Caused by: java.io.FileNotFoundException: http://localhost:8080/history/98613681-91f2-4f42-ac23-655de09178b3
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1872)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at org.http4s.client.JavaNetClientBuilder.$anonfun$fetchResponse$2(JavaNetClientBuilder.scala:128)
```
Note that the call succeeds using BlazeClientBuilder

**Comment by rossabaker** [comment]
Are you sure it's a `NoContent` rather than `NotFound`?  I can reproduce it on the latter, but not the former.

**Comment by steve-e** [comment]
> Are you sure it's a `NoContent` rather than `NotFound`? I can reproduce it on the latter, but not the former.

You're right its NotFound that failed. Thanks

**Comment by aeons** [comment]
Fixed by #2117
